examples/server_authenticate.rs

```diff
@@ -0,0 +1,84 @@
+use std::error::Error;
+
+use imap_codec::imap_types::response::{CommandContinuationRequest, Greeting, Status};
+use imap_flow::{
+    server::{ServerFlow, ServerFlowEvent, ServerFlowOptions},
+    stream::AnyStream,
+    types::CommandAuthenticate,
+};
+use tokio::net::TcpListener;
+
+#[tokio::main(flavor = "current_thread")]
+async fn main() -> Result<(), Box<dyn Error>> {
+    let (mut server, _) = {
+        let stream = {
+            let listener = TcpListener::bind("127.0.0.1:12345").await?;
+            let (stream, _) = listener.accept().await?;
+            stream
+        };
+
+        ServerFlow::send_greeting(
+            AnyStream::new(stream),
+            ServerFlowOptions::default(),
+            Greeting::ok(None, "server_authenticate (example)")?,
+        )
+        .await?
+    };
+
+    let mut current_authenticate_tag = None;
+
+    loop {
+        let event = server.progress().await?;
+        println!("{event:?}");
+
+        // We don't implement any real SASL mechanism in this example.
+        let pretend_to_need_more_data = rand::random();
```

Comment:
I don't understand this. If `pretend_to_need_more_data` is true, then `authenticate_finish` is never called. Is this a useful example?

Reply:
`pretend_to_need_more_data` is randomly assigned in the loop. What should happen is that you get a variety of different SASL flows randomly. If `pretend_to_need_more_data` is true, we send a continuation and ask for more. Then, we proceed due to the client sending `AuthenticateData` below and eventually finish when `pretend_to_need_more_data` is false,